### DISCUSSION TITLE
Potential StrongSwan and libreswan compatibility issue?

### DISCUSSION THREAD

**Discussion by JimChen111** [strongswan/strongswan#2387] [Q&A]
Hi,

The strongSwan IPSec in our env has been working for a really long time. But it broke recently and we noticed that the traffic going out from the host was not actually being encrypted and the OS was not picking up the strongSwan policy.

One difference that I noticed was in our new env, we installed libreswan as well (ipsec.service) on our host OS as a dependency of another package that we installed. I wonder if it's possible that the host OS is using libreswan config to override the strongSwan policy.

I tried `systemctl stop ipsec` to stop the libreswan service but I don't think strongSwan is picked up after.

Is there a way to check if that's the case? Can I config the OS to use strongSwan and make sure it's not overriden?

Thanks.

**Comment by tobiasbrunner** [comment]
Having multiple IKE daemons running is definitely not a good idea. So make sure you only have one active, i.e. stop and disable (`systemctl disable ...`) the services you don't need so they are not started again after a reboot, and enable (`systemctl enable ...`) the services you actually want to use.

**Reply by JimChen111** [reply]
Thanks Tobias! So I suppose we need to reboot to let the OS pick up the strongSwan IPSec policy for this case?

**Reply by tobiasbrunner** [reply]
A reboot shouldn't be necessary. You can start/stop the services and enable/disable them in the running system.

**Reply by JimChen111** [reply]
Hmm, I have tried stopping/disabling ipsec.service but it doesn't seem to work. I suppose it's either the strongSwan configs are still being ignored, or the root cause is something else...

**Reply by tobiasbrunner** [reply]
Is strongSwan's service enabled and running? You might want to start reading logs etc.

**Reply by JimChen111** [reply]
Actually, strongSwan is running, but it seems to be failing

```
systemctl status strongswan
● strongswan.service - strongSwan IPsec IKEv1/IKEv2 daemon using swanctl
   Loaded: loaded (/usr/lib/systemd/system/strongswan.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2024-08-06 09:13:34 UTC; 7h ago
  Process: 1760537 ExecStartPost=/usr/sbin/swanctl --load-all --noprompt (code=exited, status=0/SUCCESS)
 Main PID: 1760455 (charon-systemd)
   Status: "charon-systemd running, strongSwan 5.9.10, Linux $MY_VERSION"
    Tasks: 17 (limit: 2474036)
   Memory: 4.0M
   CGroup: /system.slice/strongswan.service
           └─1760455 /usr/sbin/charon-systemd

Aug 06 16:43:48 $HOST charon-systemd[1760455]: sending packet: from $IP1[500] to $IP2[500] (304 bytes)
Aug 06 16:43:59 $HOST charon-systemd[1760455]: retransmit 5 of request with message ID 0
Aug 06 16:43:59 $HOST charon-systemd[1760455]: sending packet: from $IP1[500] to $IP3[500] (304 bytes)
Aug 06 16:44:17 $HOST charon-systemd[1760455]: interface lxc4026c057dcba deactivated
Aug 06 16:44:17 $HOST charon-systemd[1760455]: fe80::5867:f0ff:fe63:cf26 disappeared from lxc4026c057dcba
Aug 06 16:44:17 $HOST charon-systemd[1760455]: interface lxc4026c057dcba deleted
Aug 06 16:44:40 $HOST charon-systemd[1760455]: creating delete job for CHILD_SA ESP/0x00000000/$IP4
Aug 06 16:44:40 $HOST charon-systemd[1760455]: CHILD_SA ESP/0x00000000/$IP4 not found for delete
Aug 06 16:44:40 $HOST charon-systemd[1760455]: giving up after 5 retransmits
Aug 06 16:44:40 $HOST charon-systemd[1760455]: establishing IKE_SA failed, peer not responding
```

**Reply by tobiasbrunner** [reply]
Seems your peer is not reachable. Also make sure you don't have the classic charon daemon running/enabled (`strongswan-starter` systemd service unit).
